pkg/restore/restore.go

```diff
@@ -1289,6 +1323,17 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		return warnings, errs
 	}
 
+	// if it should restore status, run a UpdateStatus
+	if shouldRestoreStatus {
+		obj.SetResourceVersion(createdObj.GetResourceVersion())
+		updated, err := resourceClient.UpdateStatus(obj, metav1.UpdateOptions{})
```

Comment:
I think that you will need to look up if the resource supports the status subresource otherwise for some resources this will return a 404 not found.

Reply:
I was thinking about this problem...
Ideally, we would call the discovery API to know if the resource supports status subresource without coupling with CRDs. 
We could, however, just ignore 404 errors, and maybe log a warning. This way we don't add a new call to the API, and also don't change the RBAC permissions needed to run velero's server. WDYT?

Reply:
@RafaeLeal if that's the case, does it mean this feature only works for resources whose status is subresource?  Is that an acceptable limitation?

In addition I do we have enough information in the `err` ?  LIke the name of the resources?

Reply:
Looks like this is not related to whether the status of resource is defined as sub-resource. 
To me, as long as the resource's client implements the UpdateStatus method, then it will work.

Reply:
@reasonerjt I think this is acceptable.. For me at least it seems that would be super weird to have a CRD with a "status" field that is not a subresource "status".

I believe that we have very raw info on `err`, basically the HTTP response from a 404 on Kubernetes. Which is something like
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "deployments.apps \"dont-exist\" not found",
  "reason": "NotFound",
  "details": {
    "name": "dont-exist",
    "group": "apps",
    "kind": "deployments"
  },
  "code": 404
  ```
  
  @blackpiglet since we are using the dynamic client, it will always implement the `UpdateStatus` method. The problem is the server answer itself